sessiontxn/isolation/readcommitted.go

```diff
@@ -187,13 +210,16 @@ func (p *PessimisticRCTxnContextProvider) handleAfterQueryError(queryErr error)
 	}
 
 	p.latestOracleTSValid = false
+	p.resetCheckTSInWriteStmt()
```

Comment:
If the `resetCheckTSInWriteStmt` is supposed to be called before every execution, it's better to put it in an upper entry place to avoid the possibility of missing function calls.The current such implementation places seem to be `OnStmtStart` and `OnStmtRetry` interfaces.

Reply:
The current implementation limits the affects to RC transactions, and we should reset  `CheckTSInWriteStmt` when starting a statement or retrying the statement or retrying the pessimistic execution.  
Um..., it's Can't seem to find a better place.

Reply:
As described above, the `OnStmtStart` and `OnStmtRetry` interface implementations of the `PessimisticRCTxnContextProvider` could be the places. 
Or if you think that's not suitable don't hesitate to refactor the current `provider` and `txnManager` related things and make it clear and more maintainable. 
Putting `reset` meaning functions in different places would make the code difficult to maintain and easy to make mistakes.